apps/api/src/app/inbox/usecases/snooze-notification/snooze-notification.usecase.ts

```diff
@@ -0,0 +1,225 @@
+      _id: job._id,
+      _organizationId: job._organizationId,
+      _userId: job._userId,
+    };
+
+    await this.standardQueueService.add({
+      name: job._id,
+      data: jobData,
+      groupId: job._organizationId,
+      options: { delay, attempts: this.RETRY_ATTEMPTS, backoff: { type: 'exponential', delay: 5000 } },
+    });
+  }
+
+  private async isSnoozeEnabled(command: SnoozeNotificationCommand) {
+    const isSnoozeEnabled = await this.featureFlagsService.getFlag({
+      key: FeatureFlagsKeysEnum.IS_SNOOZE_ENABLED,
+      defaultValue: false,
+      organization: { _id: command.organizationId } as OrganizationEntity,
+      environment: { _id: command.environmentId } as EnvironmentEntity,
+      user: { _id: command.subscriberId } as UserEntity,
+    });
+
+    if (!isSnoozeEnabled) {
+      throw new NotImplementedException();
+    }
+
+    // TODO: add per environment feature on/off on integration settings
+  }
+
+  private calculateDelayInMs(snoozeUntil: Date): number {
+    return snoozeUntil.getTime() - new Date().getTime();
+  }
+
+  private async validateDelayDuration(command: SnoozeNotificationCommand, delay: number) {
+    const tierLimit = await this.getTierLimit(command);
+
+    if (delay > tierLimit) {
+      throw new HttpException('Payment Required', 402);
+    }
+  }
+
+  private async getTierLimit(command: SnoozeNotificationCommand) {
+    const organization = await this.organizationRepository.findOne({
+      _id: command.organizationId,
+    });
+
+    const systemLimitMs = await this.featureFlagsService.getFlag({
+      key: FeatureFlagsKeysEnum.MAX_DEFER_DURATION_IN_MS_NUMBER,
+      defaultValue: SYSTEM_LIMITS.DEFER_DURATION_MS,
+      environment: { _id: command.environmentId },
+      organization: { _id: command.organizationId },
+    });
+
+    const isSpecialLimit = systemLimitMs !== SYSTEM_LIMITS.DEFER_DURATION_MS;
+    if (isSpecialLimit) {
+      return systemLimitMs;
+    }
+
+    const tierLimitMs = getFeatureForTierAsNumber(
+      FeatureNameEnum.PLATFORM_MAX_SNOOZE_DURATION,
+      organization?.apiServiceLevel || ApiServiceLevelEnum.FREE,
+      true
+    );
+
+    return Math.min(systemLimitMs, tierLimitMs);
+  }
+
+  private async findNotification(command: SnoozeNotificationCommand): Promise<MessageEntity> {
+    const message = await this.messageRepository.findOne({
+      _subscriberId: command.subscriberId,
+      _environmentId: command.environmentId,
+      channel: ChannelTypeEnum.IN_APP,
+      _id: command.notificationId,
+    });
+
+    if (!message) {
+      throw new NotFoundException(`Notification id: '${command.notificationId}' not found`);
+    }
+
+    return message;
+  }
+
+  private async createScheduledUnsnoozeJob(notification: MessageEntity, delay: number): Promise<JobEntity> {
+    const originalJob = await this.jobRepository.findOne({
+      _id: notification._jobId,
+      _environmentId: notification._environmentId,
+    });
```

Comment:
Retrieving the original job is not required here per say, but I need to fill the required fields that the job schema enforces for `create()` and this is the simplest and fastest option. Creating a separate collection just for snooze jobs might be overkill at this time.

I'll revisit this part at the end of the project, for now its not a top priority.

Reply:
I was thinking of not using the jobs collection at all and just scheduling it to bull directly, without touching the mongo jobs collection that is modelled after the Novu background worker logic. The message has all the information we need to see if the feature is used or if its snoozed so this feels like unnecessary.

Reply:
Thats a possibility and a preferable solution, but since I'm reusing the `StandardWorker`, it has some checks like if the job exists and then it determines if its a snooze job to execute the unsnooze usecase. 

I will revisit this part at the end of the project as part of the different queue initiative, we might create just new worker for snooze on bull or we might use different queue altogether, or something else, all viable solutions - I will just move the 4 lines where I call the usecase from StandardWorker elsewhere + remove this job retrieval and thats all.